reana_server/scheduler.py

```diff
@@ -125,67 +135,106 @@ def get_consumers(self, Consumer, channel):
+            self._fail_workflow(workflow_id, logs=error_message)
+        else:
+            workflow_submission["retry_count"] = retry_count + 1
+            workflow_submission["user_id"] = workflow_submission["user"]
+            del workflow_submission["user"]
+            current_workflow_submission_publisher.publish_workflow_submission(
+                **workflow_submission
+            )
+
+    def on_message(self, body, message):
         """On new workflow_submission event handler."""
-        workflow_submission = json.loads(workflow_submission)
-        logging.info("Received workflow: {}".format(workflow_submission))
+        workflow_submission = json.loads(body)
+        logging.info(f"Received workflow: {workflow_submission}")
+
+        workflow_submission_copy = workflow_submission.copy()
+
+        workflow_id = workflow_submission["workflow_id_or_name"]
         workflow_min_job_memory = workflow_submission.pop("min_job_memory", 0)
+
+        workflow_submission.pop("priority", None)
+        workflow_submission.pop("retry_count", None)
+
         if reana_ready(workflow_min_job_memory):
-            logging.info("Starting queued workflow: {}".format(workflow_submission))
-            workflow_submission.pop("priority", 0)
+            logging.info(f"Starting queued workflow: {workflow_id}")
             workflow_submission["status"] = "start"
+
+            retry = True
+            should_fail_on_error = True
+            started = False
+            error_message = ""
+
             try:
-                requeue = True
-                started = False
                 (
                     response,
                     http_response,
                 ) = current_rwc_api_client.api.set_workflow_status(
                     **workflow_submission
                 ).result()
                 http_response_json = http_response.json()
-                if http_response.status_code == 200:
-                    started = True
-                    logging.info(
-                        f'Workflow {http_response_json["workflow_id"]} successfully started.'
-                    )
-                else:
-                    logging.error(
-                        f"RWC returned an unexpected status code:\n {http_response_json}"
-                    )
+                started = True
+                logging.info(
+                    f'Workflow {http_response_json["workflow_id"]} successfully started.'
+                )
 
             except HTTPBadGateway as api_e:
+                retry = True
+                should_fail_on_error = True
                 logging.error(
                     "Workflow failed to start because RWC got an error while calling"
                     f"an external service (i.e. DB):\n {api_e}",
                     exc_info=True,
                 )
             except HTTPNotFound as not_found_e:
+                retry = False
+                should_fail_on_error = False
```

Comment:
I think in this case `should_fail_on_error` should be set to `True`?

Reply:
You cannot fail a workflow that doesn't exist (`deleted` -> `failed` transition is not possible). `job-status-consumer` will just ignore the message. We will need to modify `job-status-consumer` for this.

Reply:
Not sure it is worth doing for now.